web_src/js/components/RepoBranchTagDropdown.js

```diff
@@ -80,7 +111,36 @@ export function initRepoBranchTagDropdown(selector) {
             prev.selected = false;
           }
           item.selected = true;
-          window.location.href = item.url;
+          const url = (item.tag) ? this.tagURLPrefix + item.url + this.tagURLSuffix : this.branchURLPrefix + item.url + this.branchURLSuffix;
+          if (this.branchForm === '') {
+            if (item.tag) {
+              window.location.href = url;
+            } else {
+              window.location.href = url;
+            }
```

Comment:
This seems like a rather pointless condition that can be simplified to `window.location.href = url;`.

Reply:
You changed the lines above this to make the ternary...